### PR TITLE
Not latest version

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes 28879

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/advanced-scenarios.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/advanced-scenarios.md) | [aspnetcore/blazor/advanced-scenarios](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/advanced-scenarios?branch=pr-en-us-28878) |
| [aspnetcore/blazor/blazor-server-ef-core.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/blazor-server-ef-core.md) | [aspnetcore/blazor/blazor-server-ef-core](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/blazor-server-ef-core?branch=pr-en-us-28878) |
| [aspnetcore/blazor/call-web-api.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/call-web-api.md) | [aspnetcore/blazor/call-web-api](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/call-web-api?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/built-in-components.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/built-in-components.md) | [aspnetcore/blazor/components/built-in-components](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/built-in-components?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/cascading-values-and-parameters.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/cascading-values-and-parameters.md) | [aspnetcore/blazor/components/cascading-values-and-parameters](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/class-libraries.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/class-libraries.md) | [aspnetcore/blazor/components/class-libraries](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/class-libraries?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/control-head-content.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/control-head-content.md) | [aspnetcore/blazor/components/control-head-content](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/control-head-content?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/css-isolation.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/css-isolation.md) | [aspnetcore/blazor/components/css-isolation](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/css-isolation?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/data-binding.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/data-binding.md) | [aspnetcore/blazor/components/data-binding](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/data-binding?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/dynamiccomponent.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/dynamiccomponent.md) | [aspnetcore/blazor/components/dynamiccomponent](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/dynamiccomponent?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/event-handling.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/event-handling.md) | [aspnetcore/blazor/components/event-handling](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/index.md) | [aspnetcore/blazor/components/index](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/index?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/js-spa-frameworks.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/js-spa-frameworks.md) | [aspnetcore/blazor/components/js-spa-frameworks](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/js-spa-frameworks?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/layouts.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/layouts.md) | [aspnetcore/blazor/components/layouts](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/layouts?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/lifecycle.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/lifecycle.md) | [aspnetcore/blazor/components/lifecycle](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/lifecycle?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/prerendering-and-integration.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/prerendering-and-integration.md) | [aspnetcore/blazor/components/prerendering-and-integration](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/prerendering-and-integration?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/rendering.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/rendering.md) | [aspnetcore/blazor/components/rendering](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/rendering?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/templated-components.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/templated-components.md) | [aspnetcore/blazor/components/templated-components](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/templated-components?branch=pr-en-us-28878) |
| [aspnetcore/blazor/components/virtualization.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/components/virtualization.md) | [aspnetcore/blazor/components/virtualization](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/virtualization?branch=pr-en-us-28878) |
| [aspnetcore/blazor/debug.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/debug.md) | [aspnetcore/blazor/debug](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/debug?branch=pr-en-us-28878) |
| [aspnetcore/blazor/file-downloads.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/file-downloads.md) | [aspnetcore/blazor/file-downloads](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/file-downloads?branch=pr-en-us-28878) |
| [aspnetcore/blazor/file-uploads.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/file-uploads.md) | [aspnetcore/blazor/file-uploads](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/file-uploads?branch=pr-en-us-28878) |
| [aspnetcore/blazor/forms-and-input-components.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/forms-and-input-components.md) | [aspnetcore/blazor/forms-and-input-components](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/forms-and-input-components?branch=pr-en-us-28878) |
| [aspnetcore/blazor/fundamentals/configuration.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/fundamentals/configuration.md) | [aspnetcore/blazor/fundamentals/configuration](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/configuration?branch=pr-en-us-28878) |
| [aspnetcore/blazor/fundamentals/dependency-injection.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/fundamentals/dependency-injection.md) | [aspnetcore/blazor/fundamentals/dependency-injection](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/dependency-injection?branch=pr-en-us-28878) |
| [aspnetcore/blazor/fundamentals/environments.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/blazor/fundamentals/environments.md) | [aspnetcore/blazor/fundamentals/environments](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/environments?branch=pr-en-us-28878) |
| [aspnetcore/fundamentals/servers/kestrel/http3.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/fundamentals/servers/kestrel/http3.md) | [aspnetcore/fundamentals/servers/kestrel/http3](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/http3?branch=pr-en-us-28878) |
| [aspnetcore/grpc/troubleshoot.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/grpc/troubleshoot.md) | [aspnetcore/grpc/troubleshoot](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?branch=pr-en-us-28878) |
| [aspnetcore/razor-pages/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/razor-pages/index.md) | [aspnetcore/razor-pages/index](https://review.learn.microsoft.com/en-us/aspnet/core/razor-pages/index?branch=pr-en-us-28878) |
| [aspnetcore/security/cors.md](https://github.com/dotnet/AspNetCore.Docs/blob/b121fc669bb6b7889ad97a67d034ca754859904b/aspnetcore/security/cors.md) | [aspnetcore/security/cors](https://review.learn.microsoft.com/en-us/aspnet/core/security/cors?branch=pr-en-us-28878) |

</details>

This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/AspNetCore.Docs/pull/28878/checks).

### CODE REVIEW

File: aspnetcore/mvc/controllers/routing.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ uid: mvc/controllers/routing
 By [Ryan Nowak](https://github.com/rynowak), [Kirk Larkin](https://twitter.com/serpent5), and [Rick Anderson](https://twitter.com/RickAndMSFT)
 
 :::moniker range="< aspnetcore-6.0"
-[!INCLUDE [not-latest-version](~/includes/not-latest-version.md)]
+[!INCLUDE [not-latest-version](~/includes/not-latest-versionNM.md)]
 :::moniker-end
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/controllers/routing.md:15]: @wadepickett @guardrex @tdykstra  The rare Not Moniker version for those few files that need it.

File: aspnetcore/includes/not-latest-version.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
+:::moniker range="< aspnetcore-7.0"
+
 > [!NOTE]
-> This isn't the latest version of this article. To switch to the latest, use the ASP.NET Core version selector at the top of the table of contents.  
-> :::image type="content" source="~/_static/version-selector.png" alt-text="Version selector":::  
-> If the selector isn't visible in a narrow browser window, widen the window or select the vertical ellipsis (&vellip;) > **Table of contents**.  
-> :::image type="content" source="~/_static/toc-selector.png" alt-text="Table of contents selector":::  
+> This isn't the latest version of this article. To switch to the latest, use the ASP.NET Core version selector at the top of the table of contents.
+>
+> :::image type="content" source="~/blazor/_static/version-selector.png" alt-text="Version selector":::
```
Comment (guardrex) [inline aspnetcore/includes/not-latest-version.md:6]:
```suggestion
> :::image type="content" source="~/_static/version-selector.png" alt-text="Version selector":::
```

File: aspnetcore/includes/not-latest-version.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+> 
+> If the selector isn't visible in a narrow browser window, widen the window or select the vertical ellipsis (**&vellip;**) > **Table of contents**.
+>
+> :::image type="content" source="~/blazor/_static/toc-selector.png" alt-text="Table of contents selector":::
```
Comment (guardrex) [inline aspnetcore/includes/not-latest-version.md:10]:
```suggestion
> :::image type="content" source="~/_static/toc-selector.png" alt-text="Table of contents selector":::
```

**Review by guardrex — APPROVED** [review]
Just a nit on the path to the images for if you're moving them on this PR on another commit. I recommend the use of the one that I made with the blurred version and red box.

Also a nit on these INCLUDE refs. We decided many moons ago not to use the identifier, so I recommend for consistency to find and replace ...

```
[!INCLUDE [not-latest-version]
```

... with ...

```
[!INCLUDE[]
```

... and no space in it ... again, just for consistency with what's been done in the past.

However, things might not have remained stable on that over the years in the main doc set. I can only attest to the Blazor docs being on the original convention.

File: aspnetcore/includes/not-latest-versionNM.md [hunk 4]
```diff
@@ -8,5 +8,3 @@
 > If the selector isn't visible in a narrow browser window, widen the window or select the vertical ellipsis (**&vellip;**) > **Table of contents**.
 >
 > :::image type="content" source="~/blazor/_static/toc-selector.png" alt-text="Table of contents selector":::
```
Comment (guardrex) [inline aspnetcore/includes/not-latest-versionNM.md:10]:
```suggestion
> :::image type="content" source="~/_static/toc-selector.png" alt-text="Table of contents selector":::
```

Comment (guardrex) [reply aspnetcore/includes/not-latest-versionNM.md:10]: ... and there's one more there that the GH wouldn't let me submit a suggestion on.

**Review by tdykstra — APPROVED** [review]
Looks great!

**Pull request merged**